lib/upipe-modules/upipe_audio_merge.c

```diff
@@ -0,0 +1,638 @@
+ * @param uprobe structure used to raise events
+ * @param signature signature of the pipe allocator
+ * @param args optional arguments
+ * @return pointer to upipe or NULL in case of allocation error
+ */
+static struct upipe *upipe_audio_merge_alloc(struct upipe_mgr *mgr,
+                                             struct uprobe *uprobe,
+                                             uint32_t signature, va_list args)
+{
+    struct uref *flow_def;
+    struct upipe *upipe = upipe_audio_merge_alloc_flow(mgr,
+                            uprobe, signature, args, &flow_def);
+    if (unlikely(upipe == NULL))
+        return NULL;
+
+    struct upipe_audio_merge *upipe_audio_merge =
+                              upipe_audio_merge_from_upipe(upipe);
+    upipe_audio_merge_init_urefcount(upipe);
+    urefcount_init(upipe_audio_merge_to_urefcount_real(upipe_audio_merge),
+                   upipe_audio_merge_free);
+
+    upipe_audio_merge_init_upump_mgr(upipe);
+    upipe_audio_merge_init_upump(upipe);
+    upipe_audio_merge_init_output(upipe);
+    upipe_audio_merge_init_sub_mgr(upipe);
+    upipe_audio_merge_init_sub_inputs(upipe);
+    upipe_audio_merge_init_ubuf_mgr(upipe);
+
+    upipe_audio_merge->sub_flow_def = NULL;
+
+    upipe_audio_merge_store_flow_def(upipe, flow_def);
+
+    upipe_audio_merge_check_upump_mgr(upipe);
+
+    upipe_throw_ready(upipe);
+    return upipe;
+}
+
+/** @internal @This receives a provided ubuf manager.
+ *
+ * @param upipe description structure of the pipe
+ * @param flow_format amended flow format
+ * @return an error code
+ */
+static int upipe_audio_merge_check(struct upipe *upipe, struct uref *flow_format)
+{
+    struct upipe_audio_merge *upipe_audio_merge = upipe_audio_merge_from_upipe(upipe);
+    if (flow_format != NULL)
+        upipe_audio_merge_store_flow_def(upipe, flow_format);
+
+    if (upipe_audio_merge->flow_def == NULL)
+        return UBASE_ERR_NONE;
+
+    if (upipe_audio_merge->ubuf_mgr == NULL) {
+        upipe_audio_merge_require_ubuf_mgr(upipe, uref_dup(upipe_audio_merge->flow_def));
+        return UBASE_ERR_NONE;
+    }
+
+    return UBASE_ERR_NONE;
+}
+
+/** @internal @This processes control commands on an audio_merge pipe.
+ *
+ * @param upipe description structure of the pipe
+ * @param command type of command to process
+ * @param args arguments of the command
+ * @return an error code
+ */
+static int _upipe_audio_merge_control(struct upipe *upipe,
+                                     int command, va_list args)
+{
+    UBASE_HANDLED_RETURN(upipe_audio_merge_control_inputs(upipe, command, args));
+    UBASE_HANDLED_RETURN(upipe_audio_merge_control_output(upipe, command, args));
+
+    switch (command) {
+        case UPIPE_ATTACH_UPUMP_MGR:
+            upipe_audio_merge_set_upump(upipe, NULL);
+            return upipe_audio_merge_attach_upump_mgr(upipe);
+        case UPIPE_UNREGISTER_REQUEST:
+            return UBASE_ERR_NONE;
+        case UPIPE_GET_OUTPUT: {
```

Comment:
This is already handled by `UBASE_HANDLED_RETURN(upipe_audio_merge_control_output(upipe, command, args));`